group-coordinator/src/main/java/org/apache/kafka/coordinator/group/consumer/TopicMetadata.java

```diff
@@ -40,23 +45,31 @@ public class TopicMetadata {
      */
     private final int numPartitions;
 
+    /**
+     * Map of every partition to a set of its rackIds.
+     * If the rack information is unavailable, pass an empty set.
+     */
+    private final Map<Integer, Set<String>> partitionRacks;
```

Comment:
I am confused about `PartitionMetadata` class. Is it meant to be used here?

Reply:
I'll use it here, as mentioned in above comment